packages/create-astro/test/create-astro.test.js

```diff
@@ -1,46 +1,120 @@
+  });
+}
+
+// test
+const CreateAstro = suite('npm init astro');
 
 CreateAstro.before(async () => {
-  await del(fixturesDir);
-  await fs.promises.mkdir(fixturesDir);
+  // clean install dir
+  await del(FIXTURES_DIR);
+  await fs.promises.mkdir(FIXTURES_DIR);
+
+  // install all templates & deps before running tests
+  await Promise.all(
+    TEMPLATES.map(async ({ value: template }) => {
+      const templateDir = path.join(FIXTURES_DIR, template);
+      await execa('../../create-astro.mjs', [templateDir, '--template', template, '--commit', GITHUB_SHA, '--force-overwrite'], {
+        cwd: FIXTURES_DIR,
+      });
+      await execa('yarn', ['--frozen-lockfile', '--silent'], { cwd: templateDir });
+    })
+  );
 });
 
-for (const { value: template } of TEMPLATES) {
-  // TODO: Unskip once repo is made public. Because the repo is private, the templates can't yet be downloaded.
-  CreateAstro.skip(template, async () => {
-    const testDirectory = path.join(fixturesDir, template);
-    const { stdout } = await execa('../../create-astro.mjs', [testDirectory, '--template', template, '--force-overwrite'], { cwd: path.join(cwd, 'fixtures') });
+// enforce MAX_TEST_TIME
+CreateAstro.before.each(({ __test__ }) => {
+  if (TIMER[__test__]) throw new Error(`Test "${__test__}" already declared`);
+  TIMER[__test__] = setTimeout(() => {
+    throw new Error(`"${__test__}" did not finish within allowed time`);
+  }, MAX_TEST_TIME);
+});
+CreateAstro.after.each(({ __test__ }) => {
+  clearTimeout(TIMER[__test__]);
+});
 
-    console.log(stdout);
-    // test: path should formatted as './{dirName}'
-    assert.not.match(stdout, '././');
+for (let n = 0; n < TEMPLATES.length; n++) {
+  const template = TEMPLATES[n].value;
+  const templateDir = path.join(FIXTURES_DIR, template);
 
+  CreateAstro(`${template} (install)`, async () => {
     const DOES_HAVE = ['.gitignore', 'package.json', 'public', 'src'];
-    const DOES_NOT_HAVE = ['meta.json', 'node_modules', 'yarn.lock'];
+    const DOES_NOT_HAVE = ['.git', 'meta.json'];
 
     // test: template contains essential files & folders
     for (const file of DOES_HAVE) {
-      console.log(path.join(testDirectory, file));
-      assert.ok(fs.existsSync(path.join(testDirectory, file)), `has ${file}`);
+      assert.ok(fs.existsSync(path.join(templateDir, file)), `missing ${file}`);
     }
 
     // test: template DOES NOT contain files supposed to be stripped away
     for (const file of DOES_NOT_HAVE) {
-      assert.not.ok(fs.existsSync(path.join(testDirectory, file)), `does not have ${file}`);
+      assert.not.ok(fs.existsSync(path.join(templateDir, file)), `failed to clean up ${file}`);
+    }
+  });
+
+  CreateAstro(`${template} (dev)`, async () => {
+    // start dev server
+    const port = 3000 + n; // start new port per test
+    const devServer = execa('yarn', ['start', '--port', port], { cwd: templateDir });
+    await new Promise((resolve) => {
+      setTimeout(() => resolve(), 15000);
```

Comment:
Since we didn’t want to use a full browser test suite, we’re just giving the dev server 15 seconds to start up before running the test.